pkg/component/worker/containerd.go

```diff
@@ -148,6 +158,69 @@ func (c *ContainerD) setupConfig() error {
 	return file.WriteContentAtomically(confPath, output.Bytes(), 0644)
 }
 
+func (c *ContainerD) watchDropinConfigs(ctx context.Context) {
+	log := logrus.WithField("component", "containerd")
+	watcher, err := fsnotify.NewWatcher()
+	if err != nil {
+		log.WithError(err).Error("failed to create watcher for drop-ins")
+		return
+	}
+	defer watcher.Close()
+
+	err = watcher.Add(importsPath)
+	if err != nil {
+		log.WithError(err).Error("failed to watch for drop-ins")
+		return
+	}
+
+	debouncer := debounce.Debouncer[fsnotify.Event]{
+		Input:   watcher.Events,
+		Timeout: 3 * time.Second,
+		Filter: func(item fsnotify.Event) bool {
+			switch item.Op {
+			case fsnotify.Create, fsnotify.Remove, fsnotify.Write, fsnotify.Rename:
+				return true
+			default:
+				return false
+			}
+		},
+		Callback: func(fsnotify.Event) { c.restart() },
+	}
+
+	// Consume and log any errors from watcher
+	go func() {
+		for {
+			err, ok := <-watcher.Errors
+			if !ok {
+				return
+			}
+			log.WithError(err).Error("error while watching drop-ins")
+		}
+	}()
+
+	log.Infof("started to watch events on %s", importsPath)
+
+	err = debouncer.Run(ctx)
+	if err != nil {
+		log.WithError(err).Warn("dropin watch bouncer exited with error")
+	}
+}
+
+func (c *ContainerD) restart() {
+	log := logrus.WithFields(logrus.Fields{"component": "containerd", "phase": "restart"})
+
+	log.Info("restart requested")
+	if err := c.setupConfig(); err != nil {
+		log.WithError(err).Warn("failed to resolve config")
+		return
+	}
+
+	p := c.supervisor.GetProcess()
+	if err := p.Signal(syscall.SIGHUP); err != nil {
```

Comment:
By sending SIGHUP we're killing the process rather than reloading it, so I would send SIGTERM but it's a nitpick. If you agree and want to change it ping me on slack after the change.

Reply:
I guess my rationale for `SIGHUP` was that it is the "standard" for reloading. In containerd there's no realoading support per se, the only way is to restart the process. But having `SIGHUP` do that kinda makes sense, especially if containerd ever implement actual reload using `SIGHUP`. Based on the comments in https://github.com/containerd/containerd/issues/4478, it seems there's some "plans" to actually implement reloading.

TL;DR; I'd keep it as `SIGHUP` for "forward compatibility"

Reply:
OK that's fair enough.